shared/src/commonMain/kotlin/org/hyperskill/app/track/injection/TrackComponentImpl.kt

```diff
@@ -18,6 +14,8 @@ class TrackComponentImpl(private val appGraph: AppGraph) : TrackComponent {
             appGraph.buildTopicsDataComponent().topicsInteractor,
             appGraph.analyticComponent.analyticInteractor,
             appGraph.sentryComponent.sentryInteractor,
-            urlPathProcessor
+            appGraph.buildMagicLinksDataComponent().urlPathProcessor,
+            appGraph.buildGamificationToolbarComponent().gamificationToolbarReducer,
+            appGraph.buildGamificationToolbarComponent().gamificationToolbarActionDispatcher
```

Comment:
 I guess buildGamificationToolbarComponent() should not be called twice.
 Let's move it's creation upper to store in a variable.